server/src/test/java/io/crate/planner/UnionPlannerTest.java

```diff
@@ -279,4 +283,33 @@ public void test_union_works_when_both_sides_have_no_outputs_to_keep_on_prunning
         );
     }
 
+    /*
+     * https://github.com/crate/crate/issues/17341
+     */
+    @UseRandomizedOptimizerRules(alwaysKeep = {MoveOrderBeneathUnion.class, MoveOrderBeneathEval.class, RemoveRedundantEval.class})
```

Comment:
The annotation `@UseRandomizedOptimizerRules` is only applicable in Integration Tests. Here, it has no effect. So we can remove it.

Reply:
So when using `SQLExecutor` all optimizer rules are always used?